PINCache/PINDiskCache.m

```diff
@@ -243,24 +269,31 @@ +(BOOL)moveItemAtURLToTrash:(NSURL *)itemURL
 
 + (void)emptyTrash
 {
-    PINBackgroundTask *task = [PINBackgroundTask start];
-    
-    dispatch_async([self sharedTrashQueue], ^{
-        NSError *searchTrashedItemsError = nil;
-        NSArray *trashedItems = [[NSFileManager defaultManager] contentsOfDirectoryAtURL:[self sharedTrashURL]
-                                                              includingPropertiesForKeys:nil
-                                                                                 options:0
-                                                                                   error:&searchTrashedItemsError];
-        PINDiskCacheError(searchTrashedItemsError);
-        
-        for (NSURL *trashedItemURL in trashedItems) {
-            NSError *removeTrashedItemError = nil;
-            [[NSFileManager defaultManager] removeItemAtURL:trashedItemURL error:&removeTrashedItemError];
-            PINDiskCacheError(removeTrashedItemError);
-        }
-        
+    id<PINBackgroundTask> task = [PINApplicationBackgroundTask start];
+    [self emptyTrashWithCompletion:^{
         [task end];
-    });
+    }];
+}
+
++ (void)emptyTrashWithCompletion:(void (^)(void))completion
+{
+  dispatch_async([self sharedTrashQueue], ^{
+    NSError *searchTrashedItemsError = nil;
+    NSArray *trashedItems = [[NSFileManager defaultManager] contentsOfDirectoryAtURL:[self sharedTrashURL]
+                                                          includingPropertiesForKeys:nil
+                                                                             options:0
+                                                                               error:&searchTrashedItemsError];
+    PINDiskCacheError(searchTrashedItemsError);
+    
+    for (NSURL *trashedItemURL in trashedItems) {
+      NSError *removeTrashedItemError = nil;
+      [[NSFileManager defaultManager] removeItemAtURL:trashedItemURL error:&removeTrashedItemError];
+      PINDiskCacheError(removeTrashedItemError);
```

Comment:
Indentation here seems to have changed from 4-space to 2-space, might want to re-indent for consistency.